compiler/src/dotty/tools/repl/Rendering.scala

```diff
@@ -43,26 +43,29 @@ private[repl] class Rendering(compiler: ReplCompiler,
       myClassLoader
     }
 
-  /** Load the value of the symbol using reflection
+  /** Load the value of the symbol using reflection.
    *
    *  Calling this method evaluates the expression using reflection
    */
   private[this] def valueOf(sym: Symbol)(implicit ctx: Context): Option[String] = {
     val defn = ctx.definitions
-    val objectName = sym.owner.fullName.encode.toString.dropRight(1) // gotta drop the '$'
+    val objectName = sym.owner.fullName.encode.toString.stripSuffix("$")
     val resObj: Class[_] = Class.forName(objectName, true, classLoader())
-
-    val res =
+    val value =
       resObj
-        .getDeclaredMethods.find(_.getName == sym.name.toString + "Show").get
-        .invoke(null).toString
-
+        .getDeclaredMethods.find(_.getName == sym.name.encode.toString).get
+        .invoke(null)
+    val string = value match {
+      case null => "null" // Calling .toString on null => NPE
+      case ""   => "\"\"" // Sepcial cased for empty string, following scalac
+      case x    => x.toString
+    }
     if (!sym.is(Flags.Method) && sym.info == defn.UnitType)
       None
-    else if (res.startsWith(str.REPL_SESSION_LINE))
-      Some(res.drop(str.REPL_SESSION_LINE.length).dropWhile(c => c.isDigit || c == '$'))
+    else if (string.startsWith(str.REPL_SESSION_LINE))
+      Some(string.drop(str.REPL_SESSION_LINE.length).dropWhile(c => c.isDigit || c == '$'))
```

Comment:
I know the code above is not new, but I'm wondering why this is needed

Reply:
Ok I get it. It is to pretty print a class. However it quickly leaks:
```scala
scala> class Foo 
// defined class Foo
scala> new Foo 
val res0: Foo = Foo@7baf1f5a
scala> println(res0.toString) 
rs$line$1$Foo@7baf1f5a
```
It is the same in `scalac`, so I guess that's fine. I imagine we could generate a default `toString` for classes in the REPL, but it is maybe not worth the trouble